src/utils/useSessionCount.ts

```diff
@@ -0,0 +1,42 @@
+import AsyncStorage from '@react-native-community/async-storage'
+import { useEffect } from 'react'
+
+const SESSION_COUNT_KEY = '@session-count'
+
+const getSessionCount = async () => {
+  const sessionCount = await AsyncStorage.getItem(SESSION_COUNT_KEY)
```

Comment:
nit: if this returned 0 when storage was uninitialized rather than null, a lot of the code below could be simplified